src/daemon/command_parser_executor.cpp

```diff
@@ -187,17 +187,32 @@ bool t_command_parser_executor::print_block(const std::vector<std::string>& args
 
 bool t_command_parser_executor::print_transaction(const std::vector<std::string>& args)
 {
-  if (args.empty())
+  bool include_hex = false;
+  bool include_json = false;
+  bool unexpected_optional_argument = false;
+
+  for (unsigned int i = 1; i < args.size(); ++i) {
+    if (args[i] == "--with-hex")
+      include_hex = true;
+    else if (args[i] == "--with-json")
+      include_json = true;
+    else
+    {
+      unexpected_optional_argument = true;
+      break;
+    }
+  }
+  if (args.empty() || unexpected_optional_argument)
   {
-    std::cout << "expected: print_tx <transaction hash>" << std::endl;
+    std::cout << "expected: print_tx <transaction_hash> [--with-hex] [--with-json]" << std::endl << std::endl;
     return true;
   }
```

Comment:
This loop looks like it requires the options to be at the end, which is usually not the case. I'm not sure whether this is worth changing or not.

Reply:
Yes it requires them to be after the mandatory <transaction_hash> arg. Would suggest to (i) leave as is for now, (ii) implement it in a fully flexible manner if we later move all flags of all commands to be non-positional.
Let me know if you agree.

Reply:
OK